### PR TITLE
:construction_worker: Add octodns sync and apply pipelines on pull and push

### PR DESCRIPTION
**Pull request by jasonBirchall**
This commit connects to https://github.com/ministryofjustice/operations-engineering/issues/4565 and relates to the requirement to manage our dns estate in code.

### CODE REVIEW

**Review by tamsinforbes — DISMISSED** [review]
🌝🌖🚀LGTM🚀🌘🌚

File: .github/workflows/dry-run.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      - name: Install dependencies
+        run: |
+          python -m venv venv
+          source venv/bin/activate
+          pip install -r requirements.txt
+
+      - name: Run OctoDNS dry run
+        run: |
+          source venv/bin/activate
+          octodns-sync --config-file=./config/production/octodns.yaml
```
Comment (tamsinforbes) [inline .github/workflows/dry-run.yaml:30]:
```suggestion
          octodns-sync --config-file=./config/octodns.yaml
```
No production directory

File: .github/workflows/sync.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+          python -m venv venv
+          source venv/bin/activate
+          pip install -r requirements.txt
+
+      - name: Run OctoDNS sync
+        run: |
+          source venv/bin/activate
+          octodns-sync --config-file=./config/production/octodns.yaml --doit
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
```
Comment (tamsinforbes) [inline .github/workflows/sync.yaml:32]:
```suggestion
          AWS_ACCESS_KEY_ID: ${{ secrets.OCTODNS_AWS_ACCESS_KEY_ID }}
```
In the other workflow both of these are prefixed with `OCTODNS`

File: config/dsd.io.yaml
Comment (tamsinforbes) [inline config/dsd.io.yaml:1]: I'm not reading this!

File: config/octodns.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    default_ttl: 3600
+    enforce_order: True
+  route53:
+    class: octodns_route53.Route53Provider
+    access_key_id: env/AWS_ACCESS_KEY_ID
+    secret_access_key: env/AWS_SECRET_ACCESS_KEY
+
+zones:
+  dsd.io.:
+    lenient: true
```
Comment (tamsinforbes) [inline config/octodns.yaml:15]: lenient flag = "I know what I'm doing and accept any problems I run across."

Is this wise?

File: .github/workflows/dry-run.yaml (hunk 1 above)
Comment (jasonBirchall) [reply .github/workflows/dry-run.yaml]: Absolutely. It was a sloppy copy and paste error.

File: config/dsd.io.yaml
Comment (jasonBirchall) [reply config/dsd.io.yaml:1]: Agreed. Just wait for the "all hosted zone" export I'm about to do 🧌

File: config/octodns.yaml (hunk 3 above)
Comment (jasonBirchall) [reply config/octodns.yaml:15]: This has to be true as octodns demands a trailing ".", which r53 cannot cater for.

Comment (jasonBirchall) [reply config/octodns.yaml:15]: Should I add a comment?

**Review by tamsinforbes — APPROVED** [review]
🌝🌖🚀LGTM🚀🌘🌚

**Pull request merged**